packages/xrpl/test/integration/transactions/escrowCancel.test.ts

```diff
@@ -0,0 +1,112 @@
+import { assert } from 'chai'
+import { EscrowCancel, EscrowCreate, unixTimeToRippleTime } from 'xrpl-local'
+
+import serverUrl from '../serverUrl'
+import {
+  setupClient,
+  teardownClient,
+  type XrplIntegrationTestContext,
+} from '../setup'
+import { generateFundedWallet, getXRPBalance, testTransaction } from '../utils'
+
+// how long before each test case times out
+const TIMEOUT = 30000
+
+describe('EscrowCancel', () => {
+  let testContext: XrplIntegrationTestContext
+
+  beforeEach(async () => {
+    testContext = await setupClient(serverUrl)
+  })
+  afterEach(async () => teardownClient(testContext))
+
+  it(
+    'base',
+    async () => {
+      // Funding the wallet can take some time, so we do it first BEFORE getting the ledger close_time.
+      const wallet1 = await generateFundedWallet(testContext.client)
+
+      // get the most recent close_time from the standalone container for cancel & finish after.
+      const CLOSE_TIME: number = (
+        await testContext.client.request({
+          command: 'ledger',
+          ledger_index: 'validated',
+        })
+      ).result.ledger.close_time
+
+      // Attempt to get the time after which we can check for the escrow to be finished.
+      // Sometimes the ledger close_time is in the future, so we need to wait for it to catch up.
+      const currentTimeUnix = Math.floor(new Date().getTime())
+      const currentTimeRipple = unixTimeToRippleTime(currentTimeUnix)
+      const closeTimeCurrentTimeDiff = currentTimeRipple - CLOSE_TIME
+      let waitTimeInMs = Math.min(
+        Math.abs(closeTimeCurrentTimeDiff) * 1000 + 5000,
+        // Maximum wait time of 20 seconds
+        20000,
+      )
+      waitTimeInMs = waitTimeInMs < 5000 ? 5000 : waitTimeInMs
+
```

Comment:
This might be worth extracting into a helper since it's shared across two of the Escrow tests.

Reply:
Done